src/main/java/de/tum/in/www1/artemis/service/scheduled/cache/quiz/QuizCache.java

```diff
@@ -98,6 +98,8 @@ protected void clear() {
      */
     void updateQuizExercise(QuizExercise quizExercise) {
         Objects.requireNonNull(quizExercise, "quizExercise must not be null");
+        // ensure that a local cache exists to save the exercise in
+        getTransientWriteCacheFor(quizExercise.getId());
```

Comment:
This should no have any effect, since `updateQuizExerciseLocally` just below this method executes the same method.

Reply:
This is currently needed to fix the tests flakynes. Now the chache gets created in the TestWorker thread, not in the hazelcast event thread. Removing this call makes if flaky again (6 fails in 3000 runs without it, 0 fails in 9000 runs with it).

I now added an additional call to `getWriteCache()` to the test case. This has the same effect without changing the production code, but this means that the method is now package private.